ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+#         description="Is the encoder trainable.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["trainable"],
+#     )
+#
+#     use_pretrained: bool = schema_utils.Boolean(
+#         default=True,
+#         description="Use pre-trained model weights from Hugging Face.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["use_pretrained"],
+#     )
+#
+#     pretrained_model: str = schema_utils.String(
+#         default="google/vit-base-patch16-224",
+#         description="The name of the pre-trained model to use.",
+#         parameter_metadata=ENCODER_METADATA["ViTEncoder"]["pretrained_model"],
+#     )
+
+
+@dataclass
+class TVBaseEncoderConfig(BaseEncoderConfig):
+    use_pretrained: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Download model weights from pre-trained model.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["use_pretrained"],
     )
 
-    batch_norm_epsilon: float = schema_utils.NonNegativeFloat(
-        default=0.001,
-        description="Epsilon of the batch norm.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["batch_norm_epsilon"],
+    model_cache_dir: str = schema_utils.String(
+        description="Directory path to cache pretrained model weights.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["model_cache_dir"],
     )
 
-    use_bias: Optional[bool] = schema_utils.Boolean(
+    saved_weights_in_checkpoint: bool = schema_utils.Boolean(
+        default=False,
+        description="Whether to save the weights in the checkpoint.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["saved_weights_in_checkpoint"],
+    )
+
+    trainable: bool = schema_utils.Boolean(
         default=True,
-        description="Whether the layer uses a bias vector.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["use_bias"],
+        description="Is the encoder trainable.",
+        parameter_metadata=ENCODER_METADATA["TVBaseEncoder"]["trainable"],
     )
 
-    bias_initializer: Optional[str] = schema_utils.StringOptions(
-        sorted(list(initializer_registry.keys())),
-        default="zeros",
-        description="initializer for the bias vector.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["bias_initializer"],
+
+@register_encoder_config("alexnet_torch", IMAGE)
+@dataclass
+class TVAlexNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["alexnet_torch"],
+        default="alexnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    weights_initializer: Optional[str] = schema_utils.StringOptions(
-        sorted(list(initializer_registry.keys())),
-        default="xavier_uniform",
-        description="Initializer for the weights matrix.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["weights_initializer"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
done